ngraph/core/include/openvino/op/loop.hpp

```diff
@@ -30,7 +30,7 @@ class OPENVINO_API Loop : public op::util::SubGraphOp {
         int64_t body_condition_output_idx = -1;
     };
 
-    OPENVINO_OP("Loop", "opset5", op::util::SubGraphOp, 5);
+    OPENVINO_OP("Loop", "opset5", op::util::MultiSubGraphOp, 5);
```

Comment:
But it is inherited from SubGraphOp. I'm a bit confused

Reply:
yes, Loop inherits from SubGraphOp and SubGraphOp from MultiSubGraphOp, I've changed it as a result of https://github.com/openvinotoolkit/openvino/pull/8035#discussion_r756141832 comment.

Reply:
There is no difference for now, but it's better to used SubGraphOp for Loop/TI and MultiSubGraphOp for If

Reply:
Looks like RTTI in SubGraphOp is not defined correctly `OPENVINO_OP("SubGraphOp", "util");` as it doesn't mention it's pattern in RTTI hierarchy.

Reply:
And yes, we don't have to change base class in Loop RTTI.

Reply:
@GlebKazantaev So to sum up, should I restore default state: `SubGraphOp` in TI, Loop RTTI
and
```
    auto multi_subgraph_op_pattern = pattern::wrap_type<
        op::util::MultiSubGraphOp, opset8::Loop, opset8::TensorIterator, opset8::If>();
```
?

Reply:
We need to fix SubGraphOp RTTI and use SubGraphOp inside pattern.

Reply:
@GlebKazantaev  fixed